### PR TITLE
Address issues for release

### PR DESCRIPTION
**Pull request by johnhalbert**
- Fix linting errors
- Add tests for new methods on SyncEntity
- Use "disabled_chain" id instead of client id, add DisabledChainMarker type

### CODE REVIEW

**Review by AlexeyBarabash — COMMENTED** [review]
I have verified the PR works well for deleting the sync chain with a local server built from `b/11-22-22-release-check-fixes`. 
Also these changes LGTM, but please wait for the review from @yrliou

File: command/command_test.go
Comment (yrliou) [inline command/command_test.go:676]: Can you explain this change? Thanks.

File: datastore/sync_entity.go
Comment (yrliou) [inline datastore/sync_entity.go:315]: Let's make it a pre-defined constant.

File: datastore/sync_entity.go
Comment (yrliou) [inline datastore/sync_entity.go:378]: DITTO, use constant value for this instead of hard-coded.

File: datastore/sync_entity.go
Comment (yrliou) [inline datastore/sync_entity.go:442]: DITTO

File: datastore/sync_entity_test.go
Comment (yrliou) [inline datastore/sync_entity_test.go:756]: I wonder why we do separate testing for sync entities and tag items?
Maybe we should check ClearServerData would delete both in one call?

**Review by yrliou — COMMENTED** [review]
Please see my comments below and https://github.com/brave/go-sync/pull/108/files/706de94d863dbcfb3da8c3e1300f9360169aab31#r1039947854 in the original PR.

File: command/command_test.go
Comment (johnhalbert) [reply command/command_test.go:676]: I updated the protobuf schemas and the name field, among others, were removed. The original fields are now reserved in the schema, but unused.  So, the generated Go code dropped the Name field.

File: datastore/sync_entity_test.go
Comment (johnhalbert) [reply datastore/sync_entity_test.go:756]: `InsertSyncEntitiesWithServerTags` creates both tags and sync entities.  The second call to `ClearServerData` provides a test case that deletes both.

**Review by yrliou — APPROVED** [review]

**Pull request merged**
